fe/fe-core/src/main/java/com/starrocks/planner/DeltaLakeScanNode.java

```diff
@@ -115,97 +89,34 @@ protected String debugString() {
-        DeltaUtils.checkTableFeatureSupported(snapshot.getProtocol(), deltaMetadata);
-
-        String catalogName = deltaLakeTable.getCatalogName();
         Engine engine = deltaLakeTable.getDeltaEngine();
         long snapshotId = snapshot.getVersion(engine);
-        String dbName = deltaLakeTable.getDbName();
-        String tableName = deltaLakeTable.getTableName();
-        Map<PartitionKey, Long> partitionKeys = Maps.newHashMap();
 
         GetRemoteFilesParams params =
                 GetRemoteFilesParams.newBuilder().setTableVersionRange(TableVersionRange.withEnd(Optional.of(snapshotId)))
                         .setPredicate(predicate).setFieldNames(fieldNames).build();
-        List<RemoteFileInfo> splits =
-                GlobalStateMgr.getCurrentState().getMetadataMgr().getRemoteFiles(deltaLakeTable, params);
-        if (splits.isEmpty()) {
-            LOG.warn("There is no scan tasks after planFiles on {}.{} and predicate: [{}]", dbName, tableName, predicate);
-            return;
-        }
-
-        DeltaLakeRemoteFileDesc remoteFileDesc = (DeltaLakeRemoteFileDesc) splits.get(0).getFiles().get(0);
-        if (remoteFileDesc == null) {
-            LOG.warn("There is no scan tasks after planFiles on {}.{} and predicate: [{}]", dbName, tableName, predicate);
-            return;
-        }
-
-        List<FileScanTask> splitsInfo = remoteFileDesc.getDeltaLakeScanTasks();
-        for (FileScanTask split : splitsInfo) {
-            List<String> partitionValues = new ArrayList<>();
-            split.getPartitionValues().forEach((key, value) -> partitionValues.add(value));
-            PartitionKey partitionKey = PartitionUtil.createPartitionKey(partitionValues,
-                    deltaLakeTable.getPartitionColumns(), deltaLakeTable);
-            addPartitionLocations(partitionKeys, partitionKey, descTbl, split.getFileStatus(), deltaMetadata);
-        }
-
-        scanNodePredicates.setSelectedPartitionIds(partitionKeys.values());
-    }
-
-    private void addPartitionLocations(Map<PartitionKey, Long> partitionKeys, PartitionKey partitionKey,
-                                       DescriptorTable descTbl, FileStatus fileStatus, Metadata metadata) {
-        long partitionId = -1;
-        if (!partitionKeys.containsKey(partitionKey)) {
-            partitionId = nextPartitionId();
-            Path filePath = new Path(URLDecoder.decode(fileStatus.getPath(), StandardCharsets.UTF_8));
-
-            DescriptorTable.ReferencedPartitionInfo referencedPartitionInfo =
-                    new DescriptorTable.ReferencedPartitionInfo(partitionId, partitionKey,
-                            filePath.getParent().toString());
-            descTbl.addReferencedPartitions(deltaLakeTable, referencedPartitionInfo);
-            partitionKeys.put(partitionKey, partitionId);
+        RemoteFileInfoSource remoteFileInfoSource = null;
+        if (enableIncrementalScanRanges) {
```

Comment:
If we use deltalake metadata file to collect stats, the file scan task will be cached. then if we enable incremental scan_range at this time, do we need to use cached files?  If not, we will do twice job planning.

Reply:
yes for now.  but in the future, we expect to iterate file scan tasks for only once, and that's our goal.

plus right now, `enableIncrementalScanRanges` by default is false.